plugins/lang-plan-a/src/main/java/org/elasticsearch/plan/a/Analyzer.java

```diff
@@ -500,9 +503,74 @@ public Void visitReturn(final ReturnContext ctx) {
         visit(exprctx);
         markCast(expremd);
 
-        returnsmd.allExit = true;
-        returnsmd.allReturn = true;
-        returnsmd.anyReturn = true;
+        returnsmd.methodEscape = true;
+        returnsmd.loopEscape = true;
+        returnsmd.allLast = true;
+
+        returnsmd.count = 1;
+
+        return null;
+    }
+
+    @Override
+    public Void visitTry(final TryContext ctx) {
+        final StatementMetadata trysmd = adapter.getStatementMetadata(ctx);
+
+        final BlockContext blockctx = ctx.block();
+        final StatementMetadata blocksmd = adapter.createStatementMetadata(blockctx);
+        blocksmd.lastSource = trysmd.lastSource;
+        blocksmd.inLoop = trysmd.inLoop;
+        blocksmd.lastLoop = trysmd.lastLoop;
+        incrementScope();
```

Comment:
Its clearer to you because you are used to it. For me, every time I see a `visit` call I have to think "why does this one get an increment/decrement and not that one?" and "do all calls to `visit` with this parameter type increment scope?" and things like that. That second question is why I asked if you could push the increment/decrement to the implementation. Making a method call for it probably wouldn't make it clearer though.